Dev/abzu 217070 add message detail blob

JiviteshT opened the pull request.

JiviteshT committed:
Abzu-217070-Added blob storage to store message detail.

JiviteshT committed:
Abzu-217070-Updated FTs

JiviteshT committed:
Abzu-217070-added/updated builder service tests

JiviteshT committed:
Merge branch 'develop' into dev/Abzu-217070-add-message-detail-blob

JiviteshT committed:
Abzu-217070-Improved logging, registered service and fixed failing tests

JiviteshT committed:
Abzu-217070-test commit

HugoBurgess commented on UKHO.PeriodicOutputService/UKHO.BESS.API.FunctionalTests/Helpers/Extensions.cs:
```diff
@@ -74,20 +75,46 @@ public static async Task DeleteTableEntries(string? connectionString, string? ta
         /// <param name="exchangeSetStandard">Sets the value as s63 or s57 of the exchangeSetStandard as per config</param>
         /// <param name="webjobConnectionString">Sets the connectionString of the webJob</param>
         /// <param name="queueName">Sets the name of the table for queue</param>
+        /// <param name="messageDetailUri">Sets the uri of message detail blob storage</param>
```
This isn't a parameter of the method any more.

JiviteshT replied:
Removed

hukho reviewed (approved):
Looks good

JiviteshT committed:
Abzu-217070-removed parameter from method comment

JiviteshT committed:
Abzu-217070-added logs

JiviteshT committed:
Abzu-217070

JiviteshT committed:
Added GetProductVersionsFromDirectory in FileSystemHelper to streamline directory traversal and product version retrieval. Updated IFileSystemHelper interface to include the new method. Refactored GetTheLatestUpdateNumber in BuilderService to use the new method, improving efficiency and readability. Removed redundant code and old method overloads.

Enhanced LINQ usage in BuilderService for better filtering and matching. Added unit tests for GetProductVersionsFromDirectory with mock setups to ensure robust validation. Improved directory traversal logic and XML documentation. Performed general code cleanup and updated namespace imports.

JiviteshT committed:
ABzu
